### PR TITLE
FOGL-1133 - Circular import disallows many unit tests from running

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Test suite result:
```
a@a ~/Development/FogLAMP/tests/unit/python/foglamp (FOGL-1133) $ pytest
======================================================= test session starts ========================================================
platform linux -- Python 3.6.3, pytest-3.4.0, py-1.5.2, pluggy-0.6.0
rootdir: /home/a/Development/FogLAMP/tests/unit/python/foglamp, inifile:
plugins: mock-1.6.0, cov-2.5.1, asyncio-0.8.0, allure-adaptor-1.7.10, aiohttp-0.3.0
collected 466 items                                                                                                                

common/test_audit_logger.py ..........                                                                                       [  2%]
common/test_configuration_manager.py ........................................................................                [ 17%]
common/test_jqfilter.py ....                                                                                                 [ 18%]
common/test_logger.py .....                                                                                                  [ 19%]
common/test_process.py ........                                                                                              [ 21%]
common/test_service_record.py .......                                                                                        [ 22%]
common/test_statistics.py ................                                                                                   [ 26%]
common/microservice_management_client/test_microservice_management_client.py .......................                         [ 31%]
common/storage_client/test_payload_builder.py ...................................................s...........                [ 44%]
common/storage_client/test_sc_exceptions.py ..........                                                                       [ 46%]
common/storage_client/test_storage_client.py ...............                                                                 [ 50%]
common/storage_client/test_utils.py ..................                                                                       [ 53%]
plugins/common/test_plugins_common_utils.py ....                                                                             [ 54%]
plugins/north/common/test_common.py .                                                                                        [ 54%]
plugins/south/test_http_south.py ...                                                                                         [ 55%]
services/common/test_avahi.py ......                                                                                         [ 56%]
services/common/test_services_common_utils.py ....                                                                           [ 57%]
services/common/microservice_management/test_instance.py ..........                                                          [ 59%]
services/core/test_connect.py ...                                                                                            [ 60%]
services/core/test_main.py .                                                                                                 [ 60%]
services/core/api/test_audit.py .................                                                                            [ 64%]
services/core/api/test_backup_restore.py ...................................                                                 [ 71%]
services/core/api/test_browser_assets.py ..........................................                                          [ 80%]
services/core/api/test_common_ping.py ...                                                                                    [ 81%]
services/core/api/test_statistics_api.py ........                                                                            [ 83%]
services/core/interest_registry/test_interest_registry.py ........                                                           [ 84%]
services/core/scheduler/test_scheduler_entities.py .........                                                                 [ 86%]
services/core/scheduler/test_scheduler_exceptions.py .....                                                                   [ 87%]
services/core/service_registry/test_exceptions.py .....                                                                      [ 89%]
services/core/service_registry/test_monitor.py ..                                                                            [ 89%]
services/core/service_registry/test_service_registry.py .........                                                            [ 91%]
services/south/test_services_south_main.py .                                                                                 [ 91%]
services/south/test_services_south_server.py ...............                                                                 [ 94%]
tasks/purge/test_purge.py ................                                                                                   [ 98%]
tasks/purge/test_purge_main.py .                                                                                             [ 98%]
tasks/statistics/test_statistics_history.py ......                                                                           [ 99%]
tasks/statistics/test_statistics_main.py .                                                                                   [100%]

============================================== 465 passed, 1 skipped in 36.55 seconds ==============================================

```

### CODE REVIEW

File: python/foglamp/services/core/service_registry/service_registry.py [hunk 1]
```diff
@@ -87,7 +86,9 @@ def _remove_from_scheduler_records(cls, service_name):
         if service_name in ("FogLAMP Storage", "FogLAMP Core"):
             return
 
-        future = asyncio.ensure_future(server.Server.scheduler.remove_service_from_task_processes(service_name))
+        # Require a local import in order to avoid circular import references
+        from foglamp.services.core.scheduler.scheduler import Scheduler
```
Comment (praveen-garg) [inline python/foglamp/services/core/service_registry/service_registry.py]: we should not do import like this!

File: tests/unit/python/foglamp/services/core/service_registry/test_service_registry.py [hunk 2]
```diff
@@ -104,7 +104,8 @@ def test_register_with_bad_management_port(self):
         assert excinfo.type is NonNumericPortError
         assert 0 == len(ServiceRegistry._registry)
 
-    def test_unregister(self):
+    @pytest.mark.asyncio
+    async def test_unregister(self):
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/service_registry/test_service_registry.py]: we are not passing any event_loop fixture to method

File: python/foglamp/services/core/service_registry/service_registry.py (hunk 1 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/service_registry/service_registry.py]: why not, specially in the light of circular imports?

File: tests/unit/python/foglamp/services/core/service_registry/test_service_registry.py (hunk 2 above)
Comment (amarendra-dianomic) [reply tests/unit/python/foglamp/services/core/service_registry/test_service_registry.py]: restored old code.

File: python/foglamp/services/core/service_registry/service_registry.py [hunk 3]
```diff
@@ -87,6 +86,12 @@ def _remove_from_scheduler_records(cls, service_name):
         if service_name in ("FogLAMP Storage", "FogLAMP Core"):
             return
 
+        # Require a local import in order to avoid circular import references
+        from foglamp.services.core import server
```
Comment (praveen-garg) [inline python/foglamp/services/core/service_registry/service_registry.py]: looks a sensible solution, unless we have a solid solution for circular dependency issue!
 https://docs.python.org/3/faq/programming.html#what-are-the-best-practices-for-using-import-in-a-module

**Pull request merged**